src/org/freedesktop/gstreamer/PadLinkException.java

```diff
@@ -0,0 +1,18 @@
+package org.freedesktop.gstreamer;
+
+public class PadLinkException
+    extends Exception
+{
+    public final PadLinkReturn linkResult;
```

Comment:
Make private and add a method - if going for inner class for PadLinkReturn, probably getType or getReturnType?

Reply:
My instinct is to not make things private unless making them public poses actual problems.  I have already come across an instance in the current API where something being private forced me to change the API where if something had been public I would have been able to accomplish my mission without modifying the library.

Reply:
@mutantbob sorry if that was confusing, I meant private field and *public* accessor method - I don't want to hide access to it.

Reply:
I understand that OO dudes love encapsulation, but I think it is overkill in this case.  If we want to guard against a future where this exception will become anything other than a wrapper for the PadLinkReturn value, then yes, we can add 4 lines of code for a getter.  But I'm not doing this unless you really think we need to guard against that future complexity.

Reply:
Haha!  I'm an OO dude only in the sense that I've been looking a lot at original OOP recently and know that Java really ain't it! :smile:

The reasoning is actually eager vs lazy initialization - the field forces the PadLinkReturn value to exist at creation.  I'd rather we left open the possibility to create it on request.  If at some point we're going to move towards JNA direct mapping we need to get rid of the damn type mappers!